### PR TITLE
Fix issues in scrollable pane

### PR DESCRIPTION
**Pull request by MaxLustig**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

It's possible to enter an infinite recursion in scrollable pane. notifySubscribers is called from _onWindowResize, which is called from forceLayoutUpdate (callable from the parent item). This calls Sticky::_onScrollEvent, which updates the Sticky's state. Sticky::componentDidUpdate can call scrollablePane.addStickyHeader, which calls _addSticky, which then calls notifySubscribers, infinitely recursing. The solution to this was to update Sticky::_onScrollEvent to not check the current of conditions of isStickyTop and isStickyBottom, since it was causing them to constantly flip, causing setState to constantly get called.

This, however, unfurled a different bug in ScrollablePane which the infinite loop had masked: the top sticky header wouldn't properly be placed in stickyAbove on creation of the ScrollablePane, but the constant changing of isStickyTop and isStickyBottom eventually caused it to snap in place. By getting rid of stickyContainer and moving stickyAbove and stickyBelow to siblings of the children of the pane, this is fixed (it also removed the need for resizeContainer).

#### Focus areas to test

Scrollable Pane

### CODE REVIEW

**Comment by dzearing** [comment]
@battletoilet Who should be looking at `ScrollablePane` PRs?

**Comment by dzearing** [comment]
@ThomasMichon are you ok with this?

File: packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx [hunk 1]
```diff
@@ -167,7 +167,6 @@ export class ScrollablePaneBase extends BaseComponent<IScrollablePaneProps, {}>
           }
         }, 1);
       }
-      this.notifySubscribers();
```
Comment (ThomasMichon) [inline packages/office-ui-fabric-react/src/components/ScrollablePane/ScrollablePane.base.tsx]: I think you should just move this call into the timeout.

**Comment by elliottsj** [comment]
Re-wording the description here for clarity for future reference:

It's possible to enter an infinite recursion in ScrollablePane:

```
     forceLayoutUpdate()
->   _onWindowResize()
-> * notifySubscribers()
->   Sticky::_onScrollEvent()
->   (React state update)
->   Sticky::componentDidUpdate()
->   scrollablePane.addStickyHeader()
->   _addSticky()
-> * notifySubscribers()
```

The solution to this was to update `Sticky::_onScrollEvent` to not check the current of conditions of `isStickyTop` and `isStickyBottom`, since it was causing them to constantly flip, causing `setState` to constantly get called.

This, however, unfurled a different bug in ScrollablePane which the infinite loop had masked: the top sticky header wouldn't properly be placed in `stickyAbove` on creation of the ScrollablePane, but the constant changing of `isStickyTop` and `isStickyBottom` eventually caused it to snap in place. By getting rid of `stickyContainer` and moving `stickyAbove` and `stickyBelow` to siblings of the children of the pane, this is fixed (it also removed the need for `resizeContainer`).

**Pull request merged**